packages/zilliqa-js-subscriptions/src/ws.ts

```diff
@@ -0,0 +1,197 @@
+//  This file is part of Zilliqa-Javascript-Library.
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//   This program is distributed in the hope that it will be useful,
+//   but WITHOUT ANY WARRANTY; without even the implied warranty of
+//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//   GNU General Public License for more details.
+//
+//   You should have received a copy of the GNU General Public License
+//   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import mitt from 'mitt';
+import { w3cwebsocket as W3CWebsocket } from 'websocket';
+import {
+  EventType,
+  NewBlockQuery,
+  NewEventQuery,
+  SocketConnect,
+  SocketState,
+  Unsubscribe,
+} from './types';
+
+export class WebSocketProvider {
+  public static NewWebSocket(
+    url: string,
+    options: any = {},
```

Comment:
It would be great if we can be clearer about the type of `options` rather than using `any`

Reply:
Yes, have done the modify :D